internal/cbohelper/cbohelper.go

```diff
@@ -0,0 +1,175 @@
+package cbohelper
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/coreos/go-semver/semver"
+	ignition_config_types_32 "github.com/coreos/ignition/v2/config/v3_2/types"
+	v1 "github.com/openshift/api/config/v1"
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+	iccignition "github.com/openshift/image-customization-controller/pkg/ignition"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate mockgen -source=cbohelper.go -package=cbohelper -destination=mock_cbouhelper.go
+type CBOHelperApi interface {
+	GenerateIronicConfig() (ignition_config_types_32.Config, error)
+	ConvergedFlowAvailable() bool
+}
+
+type Config struct {
+	BaremetalIronicAgentImage      string `envconfig:"IRONIC_AGENT_IMAGE" default:"registry.ci.openshift.org/openshift:ironic-agent:latest"`
```

Comment:
lets at least have as json per cpu arch

Reply:
I think default value should be from upstream registry

Reply:
You are correct, we still need to figure out how we get the ironic agent image.